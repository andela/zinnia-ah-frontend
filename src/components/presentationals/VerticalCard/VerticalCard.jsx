import React from 'react';

import { Card, Icon, Image } from 'semantic-ui-react';
import PopularAuthorItem from '../PopularAuthorItem/PopularAuthorItem';

// components

// styles
import './VerticalCard.scss';
import Title from '../Title/Title';
import Button from '../Button/Button';

const VerticalCard = () => (
  <Card>
    <Image
      src="https://res.cloudinary.com/nedy123/image/upload/v1531499318/mbppkzeehprspnbbcym8.png"
      wrapped
      ui={false}
    />
    <Card.Content>
      <Card.Header>
        <Title content="City of France" className="title-avg" />
        <Button
          className="btn-transparent"
          type="button"
          value={
            <Icon
              name="bookmark outline"
              style={{
                fontSize: '1.8rem',
                margin: '0',
              }}
            />
          }
        />
      </Card.Header>
      <Card.Description>
        There are many variations of passages of Lorem Ipsum available, but the
        majority have suffered alteration in some form, by injected.
      </Card.Description>
    </Card.Content>
    <div className="card-footer">
      <PopularAuthorItem
        key={'uiojklnmkopiuygh'}
        name={'ebenezer'}
        url={'notontwitter.com'}
        image={''}
        username={'eben'}
      />
    </div>
  </Card>
);

export default VerticalCard;
