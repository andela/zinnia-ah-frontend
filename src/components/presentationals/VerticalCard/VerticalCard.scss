@import '../../../assets/variables';

.ui.card {
  border-radius: 0;
  box-shadow: none;
  width: 100%;
}

.ui.card > .content > .header {
  display: flex;
  font-family: $lato;
  justify-content: space-between;
  margin: 0;
}

.ui.card > :first-child {
  border-radius: 0;
}

.ui.card:hover {
  box-shadow: 0 4px 4px $primary-black-opacity-2;
  cursor: pointer;
}

.ui.card > .image > img {
  border-radius: 0;
  display: block;
  max-height: 12.5rem;
  min-height: 12.5rem;
  width: 100%;
}
