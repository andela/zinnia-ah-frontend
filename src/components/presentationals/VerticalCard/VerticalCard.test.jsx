import React from 'react';
import { mount } from 'enzyme';

import VerticalCard from './VerticalCard';

describe('Vertical Card', () => {
  it('should render correctly', () => {
    const component = mount(<VerticalCard />);
    expect(component).toMatchSnapshot();
  });
});
