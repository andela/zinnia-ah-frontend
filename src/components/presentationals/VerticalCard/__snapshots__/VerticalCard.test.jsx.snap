// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vertical Card should render correctly 1`] = `
<VerticalCard>
  <Card>
    <div
      className="ui card"
      onClick={[Function]}
    >
      <Image
        as="img"
        src="https://res.cloudinary.com/nedy123/image/upload/v1531499318/mbppkzeehprspnbbcym8.png"
        ui={false}
        wrapped={true}
      >
        <div
          className="image"
        >
          <img
            src="https://res.cloudinary.com/nedy123/image/upload/v1531499318/mbppkzeehprspnbbcym8.png"
          />
        </div>
      </Image>
      <CardContent>
        <div
          className="content"
        >
          <CardHeader>
            <div
              className="header"
            >
              <Title
                className="title-avg"
                content="City of France"
              >
                <h1
                  className="title title-avg"
                >
                  City of France
                </h1>
              </Title>
              <Button
                className="btn-transparent"
                type="button"
                value={
                  <Icon
                    as="i"
                    name="bookmark outline"
                    style={
                      Object {
                        "fontSize": "1.8rem",
                        "margin": "0",
                      }
                    }
                  />
                }
              >
                <button
                  className="btn btn-transparent"
                  type="button"
                >
                  <Icon
                    as="i"
                    name="bookmark outline"
                    style={
                      Object {
                        "fontSize": "1.8rem",
                        "margin": "0",
                      }
                    }
                  >
                    <i
                      aria-hidden="true"
                      className="bookmark outline icon"
                      onClick={[Function]}
                      style={
                        Object {
                          "fontSize": "1.8rem",
                          "margin": "0",
                        }
                      }
                    />
                  </Icon>
                </button>
              </Button>
            </div>
          </CardHeader>
          <CardDescription>
            <div
              className="description"
            >
              There are many variations of passages of Lorem Ipsum available, but the majority have suffered alteration in some form, by injected.
            </div>
          </CardDescription>
        </div>
      </CardContent>
      <div
        className="card-footer"
      >
        <PopularAuthorItem
          image=""
          key="uiojklnmkopiuygh"
          name="ebenezer"
          url="notontwitter.com"
          username="eben"
        >
          <div
            className="author-slot"
          >
            <UserThumbnail
              href="#"
              image=""
              name="ebenezer"
              tag="a"
              url="notontwitter.com"
              username="eben"
            >
              <a
                className="user-thumbnail"
                href="notontwitter.com"
              >
                <Avatar
                  className=""
                  url="https://res.cloudinary.com/nedy123/image/upload/v1557149386/avatar_pupqrh.svg"
                >
                  <img
                    alt="avatar"
                    className="avatar "
                    src="https://res.cloudinary.com/nedy123/image/upload/v1557149386/avatar_pupqrh.svg"
                  />
                </Avatar>
                <div>
                  <span
                    className="name"
                  >
                     
                    ebenezer
                     
                  </span>
                  <span
                    className="info"
                  >
                     @
                    eben
                     
                  </span>
                </div>
              </a>
            </UserThumbnail>
            <Button
              className="btn-white right-small"
              type="submit"
              value="See Posts"
            >
              <button
                className="btn btn-white right-small"
                type="submit"
              >
                See Posts
              </button>
            </Button>
          </div>
        </PopularAuthorItem>
      </div>
    </div>
  </Card>
</VerticalCard>
`;
